### DISCUSSION TITLE
Extract text from cropped PDF page

### DISCUSSION THREAD

**Discussion by WallysFerreira** [py-pdf/pypdf#2319] [Q&A]
I've cropped a PDF page using mediabox.upper_left and mediabox.lower_right, now I want to extract the text of this cropped page but it gives me text of the entire page still. Visitor function doesn't seem to work also

I've tried converting it to an image and extracting the text with OCR but it's really slow compared to extracting the text off the PDF page, so I would really appreciate any help with this.

**Comment by pubpub-zz** [comment]
Visitor function should do most of the job. You may have difficulty to cut lines
I have no other solution to propose

**Comment by stefan6419846** [comment]
If do not care about the side effects like changing the render resolution of embedded images etc., you might consider running Ghostscript on the PDF file after cropping. At least this is what usually worked for me when working with `mutool poster`.
